scripts/scale-test/setup_cluster.py

```diff
@@ -0,0 +1,258 @@
+import json
+import logging
+import subprocess
+from argparse import ArgumentParser, Namespace
+from typing import List
+
+from benchmarks import juju
+from benchmarks.constants import DEFAULT_ADD_NODE_TOKEN, DEFAULT_ADD_NODE_TOKEN_TTL
+from benchmarks.models import Cluster, Unit
+from benchmarks.utils import timeit
+
+APP_NAME = "microk8s-node"
+DEFAULT_CHANNEL = "1.24/stable"
+
+
+LOG_FORMAT = "[%(asctime)s] [%(levelname)8s] --- %(message)s"
+LOG_DATEFMT = "%Y-%m-%d %H:%M:%S"
+logging.basicConfig(format=LOG_FORMAT, level=logging.INFO, datefmt=LOG_DATEFMT)
+
+
+@timeit
+def install_microk8s(
+    model, units: List[Unit], channel=DEFAULT_CHANNEL, http_proxy: str = None
+):
+    if http_proxy is not None:
+        configure_http_proxy(units, http_proxy)
+    reboot_and_wait(model)
+    install_snap(channel)
+    update_etc_hosts(units)
+
+
+def update_etc_hosts(units: List[Unit]):
+    """
+    Add entries in /etc/hosts of all units for each node in the cluster.
+    This is needed as nodes's hostnames need to be resolvable in order for
+    the add-node/join process to work.
+    """
+    logging.info("Adding units hostnames on /etc/hosts")
+    command = ";".join([f"echo {u.ip}\t{u.instance_id} >> /etc/hosts" for u in units])
+    juju.run(command, app=APP_NAME).check_returncode()
+
+
+@timeit
+def install_snap(channel: str):
+    logging.info("Installing microk8s on all units")
+    status_timeout = 60 * 20  # 20 min
+    all_commands = [
+        f"snap install microk8s --classic --channel={channel}",
+        "sudo usermod -a -G microk8s ubuntu",
+        "sudo chown -f -R ubuntu ~/.kube",
+        "sudo newgrp microk8s",
+        f"microk8s status --wait-ready --timeout={status_timeout}",
+    ]
+    command = ";".join(all_commands)
+    juju.run(command, app=APP_NAME).check_returncode()
+
+
+@timeit
+def configure_http_proxy(units: List[Unit], http_proxy: str):
+    logging.info("Configuring proxy settings on units")
+    proxy_command = ";".join(
+        [
+            f"echo HTTPS_PROXY={http_proxy} >> /etc/environment",
+            f"echo HTTP_PROXY={http_proxy} >> /etc/environment",
+            f"echo https_proxy={http_proxy} >> /etc/environment",
+            f"echo http_proxy={http_proxy} >> /etc/environment",
+        ]
+    )
+    juju.run(proxy_command, app=APP_NAME).check_returncode()
+
+    # NO_PROXY settings is unit-specific
+    for unit in units:
+        NO_PROXY = f"10.1.0.0/16,10.152.183.0/24,127.0.0.1,{unit.ip},{unit.instance_id},10.246.154.0/24"
```

Comment:
This is still an issue, because it still breaks traffic to other nodes in the cluster. Instead, I think it's best to go with this, which handles all cases, and is not blocked by waiting for the unit IP address:

```
NO_PROXY=10.0.0.0/8,127.0.0.0/8,192.168.0.0/16
```

Reply:
👍🏼 should I add the juju instance id too, or it is not needed?